webapp/src/components/Analysis/PerturbationTestingPreview.test.tsx

```diff
@@ -0,0 +1,167 @@
+    renderPerturbationTestingPreview({ train: false, eval: true });
+    await waitFor(() => {
+      //Verifying if the train toggle button is disabled if not available in datasplits
+      expect(screen.getByRole("button", { pressed: false })).toHaveClass(
+        "Mui-disabled"
+      );
+      // By default, evaluation toggle is selected on page load.
+      expect(screen.getByRole("button", { pressed: true })).not.toHaveClass(
+        "Mui-disabled"
+      );
+    });
+  });
+
+  it("should display the data in the toggle with expected styling (color, font), format", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    await waitFor(() => {
+      // Verify if the actual response data from API is visibly displayed on the toggle
+      expect(screen.getByText(/14.0%/i)).toBeVisible();
+      expect(screen.getByText("Failure rate - Evaluation Set")).toBeVisible();
+      expect(screen.getByText(/16.0%/i)).toBeVisible();
+      expect(screen.getByText("Failure rate - Training Set")).toBeVisible();
+
+      // styling should be applied for the selected toggle as expected
+      expect(screen.getByText(/14.0%/)).toHaveStyle(
+        "font-size: 3.75rem; color: rgb(237, 108, 2)"
+      );
+
+      // verify with some random unexpected values to not have been displayed.
+      expect(screen.queryByText(/17.0%/i)).toBeNull();
+      expect(screen.queryByText("some random text")).toBeNull();
+      expect(screen.queryByText(/26.0%/i)).toBeNull();
+    });
+  });
+
+  it("should not display a value if datasplit is not available", async () => {
+    renderPerturbationTestingPreview({ train: false, eval: true });
+    await waitFor(() => expect(screen.getByText("--%")).toBeVisible());
+  });
+
+  it("should have expected columns", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    const expectedColumnHeaders = [
+      "Test Family",
+      "Test Name",
+      "Modif. Type",
+      "Test Description",
+      "FR on Evaluation Set",
+    ];
+    const actualColumnHeaders = screen.getAllByRole("columnheader");
+    await waitFor(() => {
+      expectedColumnHeaders.forEach((name, index) => {
+        expect(actualColumnHeaders[index].textContent).toEqual(name);
+      });
+    });
+  });
+
+  it("should have default sort as desc for colum header 'FR on Evaluation/Training set'", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    await waitFor(() =>
+      expect(
+        screen.getAllByRole("columnheader")[4].getAttribute("aria-sort")
+      ).toBe("descending")
+    );
+  });
+
+  it("should modify the last Column 'Failure Rate' and tooltips of it if the toggle is changed to train/eval", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    // before toggle changes
+    expect(screen.getAllByRole("columnheader")[4].textContent).toBe(
+      "FR on Evaluation Set"
+    );
```

Comment:
Similarly, we can use the more straightforward `toHaveTextContent` and just check for substring "Evaluation Set" to be unaffected by how we abbreviate "Failure Rate" (in a unit test mindset).
```diff
-expect(<...>.textContent).toBe("FR on Evaluation Set");
+expect(<...>).toHaveTextContent("Evaluation Set");
```
Also, we can get the Failure Rate column directly and save it in a variable, say `header`
```suggestion
    const header = screen.getByRole("columnheader", { name: /Failure Rate/ });
    expect(header).toHaveTextContent("Evaluation Set");
```
to reuse it after the click:
```diff
-      expect(screen.getAllByRole("columnheader")[4].textContent).toBe(
-        "FR on Training Set"
-      );
+      expect(header).toHaveTextContent("Training Set");
```